### ISSUE TITLE
+= on empty list

### ISSUE THREAD

**Issue by matthiaskramm** [python/mypy#4540]
The code

```
v = []
v += [42]
```

causes mypy to report:

```
file.py:1: error: Need type annotation for 'v'
```

mypy version: `mypy 0.570-dev-8ec7046787f92bf7b00d0c70f570fcb5d75e4b53`

**Comment by ethanhs** [comment]
Mypy is very careful in that it does not deduce types as you expect in your example. If we assume from this that `v` is a `List[int]` we will end up breaking if I do `v += 4.0`. You might want to peruse [the docs on this](https://mypy.readthedocs.io/en/latest/type_inference_and_annotations.html#explicit-types-for-collections).

**Comment by gvanrossum** [comment]
I'm not so sure. We currently support
```py
v = []
v.append(42)
```
and
```py
v = []
v.extend([42])
```
What's the difference?

**Comment by ethanhs** [comment]
Ah, you are right! I seem to have missed that, sorry.

**Comment by TH3CHARLie** [comment]
```python
v = []
v += [42]
reveal_type(v)  # N: Revealed type is 'builtins.list[builtins.int*]'

x = []
x.append(42)
reveal_type(x)  # N: Revealed type is 'builtins.list[builtins.int]'
```
Using mypy 0.770+dev from master, no more error generated while mypy 0.760 produces same error. So eems like this has been fixed in some commits to 0.770
